pymc3/variational/advi.py

```diff
@@ -19,8 +22,50 @@
 
 ADVIFit = namedtuple('ADVIFit', 'means, stds, elbo_vals')
 
+def is_discreteRV(var):
+    dist = var.distribution
+
+    # Transformed distributions are continuous. (is it true?)
+    if isinstance(dist, TransformedDistribution):
+        return False
+
+    return isinstance(dist, Discrete)
```

Comment:
Here's an example of a discrete check: https://github.com/pymc-devs/pymc3/blob/9b4ee10433cf5e92a631033a71d79334b503744d/pymc3/step_methods/metropolis.py#L94 Yours might be fine too, though.